src/mame/machine/cxd8452aq.h

```diff
@@ -0,0 +1,110 @@
+// license:BSD-3-Clause
+// copyright-holders:Brice Onken
+// thanks-to:Patrick Mackinlay
+
+/*
+ * Sony CXD8452AQ WSC-SONIC3 APbus interface and Ethernet controller controller
+ *
+ * The SONIC3 is an APbus controller designed for interfacing the SONIC Ethernet
+ * controller to the APbus while providing DMA capabilities. It also manages
+ * RAM access for the SONIC. The host can write to SONIC's RAM, but not the other
+ * way around.
+ */
+
+#ifndef MAME_MACHINE_CXD8452AQ_H
+#define MAME_MACHINE_CXD8452AQ_H
+
+#pragma once
+
+class cxd8452aq_device : public device_t, public device_memory_interface
+{
+public:
+	cxd8452aq_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	void map(address_map &map);
+
+	template <typename... T>
+	void set_apbus_address_translator(T &&...args) { m_apbus_virt_to_phys_callback.set(std::forward<T>(args)...); }
+
+	auto irq_out() { return m_irq_handler.bind(); }
+	void irq_w(int state)
+	{
+		if (state)
+		{
+			m_sonic3_reg.control |= 0x1;
+		}
+		else
+		{
+			m_sonic3_reg.control &= ~0x1;
+		}
+		m_irq_check->adjust(attotime::zero);
+	}
```

Comment:
Please group configuration member functions and live member functions – this one is a live member function for setting a signal state, but you’ve got it in between configuration member functions.